internal/secrets/generic.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"context"
+	"log/slog"
+	"net/url"
+	"strings"
+
+	"github.com/tidwall/gjson"
+)
+
+const secretPrefix = "secrets."
+
+type secretAPI interface {
+	getSecretValue(string) (string, bool)
+}
+
+type createSecretsManagerFn func(ctx context.Context, logger *slog.Logger, url *url.URL) (secretAPI, error)
+
+type secretManager struct {
+	secretAPI secretAPI
+	prefix    string
+}
+
+func (s *secretManager) lookup(_ context.Context, key string) (string, bool) {
+	if !strings.HasPrefix(key, secretPrefix) {
+		return "", false
+	}
+	key = strings.TrimPrefix(key, secretPrefix)
+	parts := strings.SplitN(key, ".", 2)
```

Comment:
can we maybe comment some of this stuff so it's clearer why we have the prefix and what's being done here